segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/Ledgers.java

```diff
@@ -85,11 +98,17 @@ static LedgerHandle openFence(long ledgerId, BookKeeper bookKeeper, BookKeeperCo
      * @return A LedgerHandle for the newly opened ledger.
      * @throws DurableDataLogException If an exception occurred. The causing exception is wrapped inside it.
      */
-    static LedgerHandle openRead(long ledgerId, BookKeeper bookKeeper, BookKeeperConfig config) throws DurableDataLogException {
+    static ReadHandle openRead(long ledgerId, BookKeeper bookKeeper, BookKeeperConfig config) throws DurableDataLogException {
         try {
             return Exceptions.handleInterruptedCall(
-                    () -> bookKeeper.openLedgerNoRecovery(ledgerId, LEDGER_DIGEST_TYPE, config.getBKPassword()));
-        } catch (BKException bkEx) {
+                    () -> FutureUtils.result(bookKeeper
```

Comment:
Same question here as above.

Reply:
same answer as above, we are bloking for the operation to complete and we should handle InterruptedException